### PR TITLE
add csi nfs image to the csv

### PR DESCRIPTION
**Pull request by Rakshith-R**
CSV merger accepts --rook-csi-nfs-image
flag for csi nfsplugin image.
It can be initialized by setting
ROOK_CSI_NFS_IMAGE env variable or
it defaults to LATEST_ROOK_CSI_NFS_IMAGE
during CSV generation.

Signed-off-by: Rakshith R <rar@redhat.com>

Refer: 
- https://issues.redhat.com/browse/RHSTOR-3101
- https://issues.redhat.com/browse/RHSTOR-2379
- https://issues.redhat.com/browse/RHSTOR-2764
- https://github.com/rook/rook/pull/9927

/cc @b-ranto @nixpanic @BlaineEXE @Madhu-1 @agarwal-mudit

### CODE REVIEW

**Comment by Rakshith-R** [comment]
> This looks okay to me, but I wonder if these changes also need to include a CRD option to enable NFS in Rook. Not sure if ocs-operator's repo prefers to keep those separate or not.

This is just to pass nfs image to rook operator.

Enabling NFS csi pods will remain a manual step through `rook-ceph-operator-override` configmap similar to csiaddons and vol-rep. There's BZ/jirra open for this and candidate for 4.12.

File: hack/generate-unified-csv.sh [hunk 1]
```diff
@@ -65,4 +65,5 @@ $CSV_MERGER \
 	--manifests-directory=$BUNDLEMANIFESTS_DIR \
 	--olm-bundle-directory="$OCS_FINAL_DIR" \
 	--timestamp="$TIMESTAMP" \
-	--rook-csiaddons-image="$ROOK_CSIADDONS_IMAGE"
+	--rook-csiaddons-image="$ROOK_CSIADDONS_IMAGE" \
+	--rook-csi-nfs-image="$ROOK_CSI_NFS_IMAGE"
```
Comment (Madhu-1) [inline hack/generate-unified-csv.sh:69]: a new line is required. GitHub is complaining about no newline at the end of the file.

**Review by Madhu-1 — COMMENTED** [review]
LGTM

File: hack/generate-unified-csv.sh (hunk 1 above)
Comment (Rakshith-R) [reply hack/generate-unified-csv.sh]: fixed thanks !

**Review by Madhu-1 — APPROVED** [review]

File: hack/generate-latest-csv.sh [hunk 2]
```diff
@@ -27,5 +28,8 @@ echo -e "\tNOOBAA_CORE_IMAGE=$NOOBAA_CORE_IMAGE"
 echo -e "\tNOOBAA_DB_IMAGE=$NOOBAA_DB_IMAGE"
 echo -e "\tOCS_IMAGE=$OCS_IMAGE"
 echo -e "\tOCS_MUST_GATHER_IMAGE=$OCS_MUST_GATHER_IMAGE"
+echo -e "\tVOLUME_REPLICATION_OPERATOR_IMAGE=$VOLUME_REPLICATION_OPERATOR_IMAGE"
```
Comment (nixpanic) [inline hack/generate-latest-csv.sh:31]: `VOLUME_REPLICATION_OPERATOR_IMAGE` should not be part of this commit? Maybe a 2nd commit, or an other PR?

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: hack/generate-latest-csv.sh (hunk 2 above)
Comment (Rakshith-R) [reply hack/generate-latest-csv.sh:31]: > `VOLUME_REPLICATION_OPERATOR_IMAGE` should not be part of this commit? Maybe a 2nd commit, or an other PR?

its just a harmless echo statement, I guess its fine.

**Review by b-ranto — APPROVED** [review]

File: hack/generate-latest-csv.sh (hunk 2 above)
Comment (nixpanic) [reply hack/generate-latest-csv.sh:31]: It is not relevant what it is, harmless or not. It is unrelated to this PR, so it should not be included.

Comment (nixpanic) [reply hack/generate-latest-csv.sh:31]: Obviously, the same counts for `ROOK_CSIADDONS_IMAGE` and any other non-NFS related modification.

Comment (Rakshith-R) [reply hack/generate-latest-csv.sh:31]: done, added another commit, PTAL @nixpanic

**Review by agarwal-mudit — APPROVED** [review]

**Comment by BlaineEXE** [comment]
If we intend installation in 4.11 to be a single simple patch, we should ensure that we only need to patch the `ROOK_CSI_ENABLE_NFS` value. Do the rest of the values (below) have the right defaults we need, or should we also set these from ocs-operator? 
- ROOK_CSI_NFS_IMAGE
- CSI_NFS_PLUGIN_UPDATE_STRATEGY
- ROOK_CSI_NFS_POD_LABELS
- CSI_NFS_PROVISIONER_NODE_AFFINITY
- CSI_NFS_PROVISIONER_TOLERATIONS
- CSI_NFS_PLUGIN_NODE_AFFINITY
- CSI_NFS_PLUGIN_TOLERATIONS
- CSI_NFS_PROVISIONER_RESOURCE
- CSI_NFS_PLUGIN_RESOURCE
